cvat-ui/src/components/analytics-page/analytics-page.tsx

```diff
@@ -42,321 +43,218 @@ function handleTimePeriod(interval: DateIntervals): [string, string] {
     }
 }
 
+type InstanceType = 'project' | 'task' | 'job';
+
 function AnalyticsPage(): JSX.Element {
     const location = useLocation();
-    let instanceType = '';
-    if (location.pathname.includes('projects')) {
-        instanceType = 'project';
-    } else if (location.pathname.includes('jobs')) {
-        instanceType = 'job';
-    } else {
-        instanceType = 'task';
-    }
 
-    const [fetching, setFetching] = useState(true);
+    const requestedInstanceType: InstanceType = (() => {
+        if (location.pathname.includes('projects')) {
+            return 'project';
+        }
+        if (location.pathname.includes('jobs')) {
+            return 'job';
+        }
+        return 'task';
+    })();
+
+    const requestedInstanceID: number = (() => {
+        if (requestedInstanceType === 'project') {
+            return +useParams<{ pid: string }>().pid;
+        }
+        if (requestedInstanceType === 'job') {
+            return +useParams<{ jid: string }>().jid;
+        }
+        return +useParams<{ tid: string }>().tid;
+    })();
+
+    const [activeTab, setTab] = useState(window.location.hash.slice(1));
```

Comment:
Perhaps here and below we also need to check that the hash is supported. It may be for example "blablabla" that will be assigned to active tab below. But such tab will not exist